spellbook/models/safe/ethereum/safe_ethereum_safes.sql

```diff
@@ -0,0 +1,54 @@
+{{ 
+    config(
+        materialized='incremental',
+        alias='safes',
+        unique_key = 'address',
+        on_schema_change='fail',
+        file_format ='delta',
+        incremental_strategy='merge'
+    ) 
+}}
+
+select
+    et.from as address,
+    case 
+        when et.to = '0x8942595A2dC5181Df0465AF0D7be08c8f23C93af' then '0.1.0'
+        when et.to = '0xb6029ea3b2c51d09a50b53ca8012feeb05bda35a' then '1.0.0'
+        when et.to = '0xae32496491b53841efb51829d6f886387708f99b' then '1.1.0'
+        when et.to = '0x34cfac646f301356faa8b21e94227e3583fe3f5f' then '1.1.1'
+        when et.to = '0x6851d6fdfafd08c0295c392436245e5bc78b0185' then '1.2.0'
+    end as creation_version,
+    et.block_time as creation_time,
+    et.tx_hash
+from {{ source('ethereum', 'traces') }} et 
+where et.success = true
+    and et.call_type = 'delegatecall' -- the delegate call to the master copy is the Safe address    
+    and (
+            (substring(et.input, 0, 10) = '0x0ec78d9e' -- setup methods of v0.1.0
+                and et.to = '0x8942595A2dC5181Df0465AF0D7be08c8f23C93af') -- mastercopy v0.1.0
+            or
+            (substring(et.input, 0, 10) = '0xa97ab18a' -- setup methods of v1.0.0
+                and et.to = '0xb6029ea3b2c51d09a50b53ca8012feeb05bda35a') -- mastercopy v1.0.0
+            or
+            (substring(et.input, 0, 10) = '0xb63e800d' -- setup methods of v1.1.0, v1.1.1, 1.2.0
+                and et.to in (
+                    '0xae32496491b53841efb51829d6f886387708f99b',  -- mastercopy v1.1.0
+                    '0x34cfac646f301356faa8b21e94227e3583fe3f5f',  -- mastercopy v1.1.1
+                    '0x6851d6fdfafd08c0295c392436245e5bc78b0185')  -- mastercopy v1.2.0
+                )
+        )
+    and et.gas_used > 0  -- to ensure the setup call was successful
+    {% if is_incremental() %}
+    and et.block_time > (select max(creation_time) from {{ this }}) - interval '1 day'
+    {% endif %}
+        
+union all
+    
+select contract_address as address, 
+    '1.3.0' as creation_version, 
+    evt_block_time as creation_time, 
+    evt_tx_hash as tx_hash
+from gnosis_safe_ethereum.GnosisSafev1_3_0_evt_SafeSetup
```

Comment:
the goal is to avoid hardcoding table names. can you build a new sources YAML file in this directory for 'gnosis_safe' specific decoded tables? you can mimic an existing sources file:
https://github.com/duneanalytics/abstractions/blob/master/spellbook/models/transfers/ethereum/transfers_ethereum_sources.yml

you can leave out column descriptions if they are obvious based on column name. that's more a nice-to-have and helps with auto-document generation for other users.

with that said, would it make more sense to rename this whole directory to 'gnosis_safe' rather than 'safe'? i ask due to the schema in the db being 'gnosis_safe_ethereum' (side note: can have 'ethereum' directory nested in too, allows for more blockchains later on).

Reply:
once built, this hardcoded table reference can use the '{{ source ( ) }}' syntax, similar to {{ref( )}} i've seen used in this PR

Reply:
Hi @jeff-dude!

Thank you for the feedback. I am back from a vacation and will adjust the PR today.
We rebranded, and now the project name is `Safe`. It would be good to deprecate all the `gnosis_safe_` names with time and change them to `safe_`. I don't know how to make it smooth, so I will use already existing decoded tables for now. Let's discuss in Discord how we could migrate properly!

>  can have 'ethereum' directory nested in too

I already have an 'ethereum' directory nested in the 'safe' directory, or did you mean some other place?

Reply:
I made adjustments @jeff-dude

Reply:
thank you!